homeassistant/components/homekit/__init__.py

```diff
@@ -357,12 +369,24 @@ def remove_bridge_accessory(self, aid):
             acc = self.bridge.accessories.pop(aid)
         return acc
 
-    def start(self, *args):
+    async def async_start(self, *args):
         """Start the accessory driver."""
         if self.status != STATUS_READY:
             return
         self.status = STATUS_WAIT
 
+        ent_reg = await entity_registry.async_get_registry(self.hass)
+
+        bridged_states = []
+        for state in self.hass.states.async_all():
+            if not state or not self._filter(state.entity_id):
+                continue
+            _async_configure_linked_battery_sensors(ent_reg, self._config, state)
```

Comment:
Can we make this smarter? Now we're iterating over full device registry for each state. `O(stateCount * deviceRegistryCount)`

If we make a lookup table of devices and their device classes, we can run it in `O(stateCount + deviceRegistryCount)`.

Reply:
As long as accessing `ent_reg.entities` directly is ok, I can definitely make this much more efficient.

Reply:
Yes that's ok. 

I am wondering here if we could make a reusable method to make a device lookup table with this structure:

```python
lookup[device_id][(domain, device_class)] = entity_id
```

Reply:
@balloob I've done that. I'm not sure I've put it in the right place but its working as you suggest.